### PR TITLE
DAOS-3414 mgmt: implement killrank drpc handler

### PR DESCRIPTION
**Pull request by tanabarr**
Trigger kill signal to self inside dRPC KillRank handler in mgmt
module. Naiive implementation to be improved in future iterations.
Clean up by removing unnecessary/unused system/mgmt pb-c files.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/mgmt/srv_drpc.c [hunk 1]
```diff
@@ -90,8 +91,15 @@ ds_mgmt_drpc_kill_rank(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
...
-
+	/* terminate local service */
+	if (req->force)
+		sig = SIGKILL;
+	else
+		sig = SIGTERM;
+	D_INFO("Service rank %d is being killed by signal %d\n",
+		req->rank, sig);
+	kill(getpid(), sig);
+}
```
Comment (kjacque) [inline src/mgmt/srv_drpc.c]: Oops?

File: src/mgmt/srv_drpc.c [hunk 2]
```diff
@@ -90,8 +91,15 @@ ds_mgmt_drpc_kill_rank(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
...
 	mgmt__daos_resp__init(resp);
 
-	/* TODO: do something with request and populate daos response status */
-
+	/* terminate local service */
+	if (req->force)
+		sig = SIGKILL;
+	else
+		sig = SIGTERM;
+	D_INFO("Service rank %d is being killed by signal %d\n",
```
Comment (kjacque) [inline src/mgmt/srv_drpc.c]: Don't we need to look up the specified rank and send a message to that node? Perhaps I am missing something.

Comment (tanabarr) [reply src/mgmt/srv_drpc.c]: it's node local at this point

Comment (tanabarr) [reply src/mgmt/srv_drpc.c]: will be executing on the rank in the request

**Comment by tanabarr** [comment]
Running 2 node (boro-44,boro-45) DAOS system in pmixless mode, list membership, run system stop, verify node stopped by querying membership again.

```
anabarr@boro-45:~/projects/daos_m$ install/bin/daos_shell -l boro-45:10001 -i system member-query
Active connections: [boro-45:10001]
System-member-query command SUCCEEDED: 10.7.1.44:10001/d578d906-7cc4-45a4-a07c-248267308666/1
tanabarr@boro-45:~/projects/daos_m$ install/bin/daos_shell -l boro-45:10001 -i system stop
Active connections: [boro-45:10001]
System-stop command SUCCEEDED: : still 1 active members
tanabarr@boro-45:~/projects/daos_m$ install/bin/daos_shell -l boro-45:10001 -i system member-query
Active connections: [boro-45:10001]
System-member-query command SUCCEEDED: no joined members
```

**Comment by tanabarr** [comment]
> Why does KillRank require a dRPC? E.g., why not kill daos_io_server from daos_server directly?

It doesn't necessarily and one of the next iterations will be to manage the instances (daos_io_server) from the harness (daos_server) and probably won't use dRPC. This PR is just to enable the basic functionality.

File: src/proto/mgmt/srv.proto [hunk 3]
```diff
@@ -27,28 +27,59 @@ package mgmt;
...
 
 // SetRankResp is identical to DaosResp.
 
 message CreateMsReq {
-	bool bootstrap = 1;
-	// Server UUID of this MS replica.
-	string uuid = 2;
-	// Server management address of this MS replica.
-	string addr = 3;
+	bool bootstrap = 1;	// Bootstrap the DAOS system (original MS leader).
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/proto/mgmt/srv.proto]: I will revise the text

**Comment by liw** [comment]
> > Why does KillRank require a dRPC? E.g., why not kill daos_io_server from daos_server directly?
> 
> It doesn't necessarily and one of the next iterations will be to manage the instances (daos_io_server) from the harness (daos_server) and probably won't use dRPC. This PR is just to enable the basic functionality.

OK, it's up to you; I thought modifying the harness would be simpler.

**Comment by liw** [comment]
ior small failed with

    23:19:02 DEBUG| [stderr] ior: src/cart/crt_rpc.c:562: crt_req_fill_tgt_uri: Assertion `base_uri != ((void *)0)' failed.

**Comment by tanabarr** [comment]
> > > Why does KillRank require a dRPC? E.g., why not kill daos_io_server from daos_server directly?
> > 
> > 
> > It doesn't necessarily and one of the next iterations will be to manage the instances (daos_io_server) from the harness (daos_server) and probably won't use dRPC. This PR is just to enable the basic functionality.
> 
> OK, it's up to you; I thought modifying the harness would be simpler.

we already had the DRPC plumbing in place so it was a quick implementation

File: src/mgmt/srv_drpc.c [hunk 4]
```diff
@@ -64,35 +66,43 @@ pack_daos_response(Mgmt__DaosResp *daos_resp, Drpc__Response *drpc_resp)
...
+		req->rank, sig);
+	kill(getpid(), sig);
 
-	mgmt__daos_rank__free_unpacked(req, NULL);
+	mgmt__kill_rank_req__free_unpacked(req, NULL);
 	pack_daos_response(resp, drpc_resp);
```
Comment (mjmac) [inline src/mgmt/srv_drpc.c]: This never gets sent, right? Does that confuse anything on the control plane side?

**Pull request merged**